src/Plutarch/Extra/List.hs

```diff
@@ -209,3 +238,132 @@ pfromList ::
   [Term s a] ->
   Term s (list a)
 pfromList = foldr (\x xs -> pcons # x # xs) pnil
+
+{- | Check if two lists are equal, given a custom equalator. The two list types
+     can be different.
+
+     @since 3.14.1
+-}
+plistEqualsBy ::
```

Comment:
Maybe use `PComparator`, but it's really fine either way.

Reply:
Hmmm, in agora's use case, the element might not have `POrd`.

Reply:
You can construct `PComparator`s without needing an `Ord` instance. I agree with @SeungheonOh here: `PComparator`s at least force you to be honest.